oap-server/server-storage-plugin/storage-influxdb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/influxdb/InfluxClient.java

```diff
@@ -101,11 +113,10 @@ private InfluxDB getInflux() {
 
         try {
             QueryResult result = getInflux().query(new Query(query.getCommand()));
-            if (result.hasError()) {
-                throw new IOException(result.getError());
-            }
+            healthChecker.health();
```

Comment:
I have checked the source code of [influx client](https://github.com/influxdata/influxdb-java/blob/bfbbcd0f4ec8bb1a6362e7a36b62d81e13693393/src/main/java/org/influxdb/impl/InfluxDBImpl.java#L804), 
when some errors happens it will throw an exception, no need to check the hasError method

Reply:
@dmsolr Could you confirm the status? Is there some case there could be some errors w/o exception?

Reply:
hi @xbkaishui 
I had test some cases. The result show following
1. syntax error/parse error, connection, etc will throw an exception.
2. aggregate function error maybe doesn't throw.

For example:
1. case 1
 ```
Query: SELECT mean("square"),"round" FROM "peg"
Response: QueryResult [results=[Result [series=null, error=mixing aggregate and non-aggregate queries is not supported]], error=null]
```

2. case 2 
```
Query: SELECT sum("square", 2),"round" FROM "peg"

Response: [results=[Result [series=null, error=invalid number of arguments for mean, expected 1, got 2]], error=null]
```

These two examples put error messages in clauses only. :(

Reply:
I am not sure that is wrong what I did before. But we need to consider this scenario, right?

Reply:
Nop, for below sql , in the java client it will throws an exception.  did you use the java client to test ?

Reply:
please check me test code [gist](https://gist.github.com/xbkaishui/493009dcb56dd9cda8b8ddd82f53f34f.js)

Reply:
@xbkaishui I am curious if all things go through exception, why InfluxDB Java client designs that API? In case the exception has been caught?

Reply:
from what I test, the error msg is wrapper as an exception in java client

Reply:
Maybe other client has different implement.

Reply:
> from what I test, the error msg is wrapper as an exception in java client

OK, if so, that we may be able to remove the codes. But in another hand, to do the check seems harmless :)

Reply:
I've read your code. I think have some issue.

```java
public static void main(String[] args) {
    final InfluxDB influxDB = InfluxDBFactory.connect("http://127.0.0.1:8086");
    String databaseName = "NOAA_water_database";
    try {
        QueryResult result = influxDB.query(new Query("SELECT sum(\"square\", 2),\"round\" FROM \"peg\"\n" +
                                                          ";" + databaseName)); // There is syntax issue, so InfluxDB Client throws an exception. You don't need to append `+ ";" + databaseName`.
        String error = result.getError();
        System.out.println(error);
        influxDB.setDatabase(databaseName);
    } catch (Throwable e) {
        e.printStackTrace();
    }
}
```

You could try following code.

```java
try {
    influxDB.setDatabase(databaseName);
    QueryResult result = influxDB.query(new Query("SELECT sum(\"square\", 2),\"round\" FROM \"peg\"\n"));
    result.hasError(); // it is false.
    result.getError(); // null

    result.getResults().forEach(series -> {
        series.hasError(); // it is true
        series.getError(); // error message
    });
    
    influxDB.setDatabase(databaseName);
} catch (Throwable e) {
    e.printStackTrace();
}
```

Reply:
Syntax errors can be detected by the client, so the exception is thrown directly.
Some errors, like aggregate function wrong, do not throw the exception.

Reply:
> Syntax errors can be detected by the client, so the exception is thrown directly.
> Some errors, like aggregate function wrong, do not throw the exception.

but for the aggregate function, the hasError return false, it will not throws an exception using the orginal code, so the check error logic is take no effect  right ?

Reply:
I will revert my changes. thanks

Reply:
done

Reply:
> but for the aggregate function, the hasError return false, it will not throws an exception using the orginal code, so the check error logic is take no effect right ?

It is basically right. Would you like to fix it in this PR?

Reply:
Do you mean also check the child resuts error ?